File: data structures and algorithms/java/05_034.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
import java.math.*;

public class Main {
    public static void main(String[] args) {
        // Main method code goes here
    }
    
    public class Graph {
        private Map<Object, List<Object>> graph;
        
        public Graph() {
            this.graph = new HashMap<>();
        }
        
        public void addVertex(Object vertex) {
            if (!this.graph.containsKey(vertex)) {
                this.graph.put(vertex, new ArrayList<>());
            }
        }
    }
}

File: data structures and algorithms/cs/05_034.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Graph
{
    private Dictionary<object, List<object>> graph;
    
    public Graph()
    {
        this.graph = new Dictionary<object, List<object>>();
    }
    
    public void AddVertex(object vertex)
    {
        if (!this.graph.ContainsKey(vertex))
        {
            this.graph[vertex] = new List<object>();
        }
    }

    public bool HasVertex(object vertex)
    {
        return this.graph.ContainsKey(vertex);
    }

    public IEnumerable<object> GetAdjacent(object vertex)
    {
        return HasVertex(vertex) ? this.graph[vertex] : Enumerable.Empty<object>();
    }
}
